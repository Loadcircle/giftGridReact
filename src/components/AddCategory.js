import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    //si useState() se deja vacio, el valor inicial sera undefined, asi que utilizar un string vacio
    const [inputValue, setInputValue] = useState(''); 

    const handleInputChange = (e)=>{
        if(inputValue.length < 15){
            setInputValue(e.target.value)
        }
    }

    const handleSubmit = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            
            //aca llamamos el callback que enviamos
            setCategories(cat=>[inputValue]);
            setInputValue('')
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}