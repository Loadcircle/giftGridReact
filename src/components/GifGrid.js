import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GridItem } from './GridItem';

export const GifGrid = ({category}) => {

    //para cambiar el nombre de data a images
    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 key={category}>{category}</h3>
            {/* esto es un operador ternario simple, si el primero es cierto muestra */}
            {loading && <p>Loading...</p> }
            <div className="card_wrapper">
                {
                    images.map(image=>(
                        <GridItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}
