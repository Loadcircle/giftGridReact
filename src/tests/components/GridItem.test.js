import React from 'react';
const { shallow } = require("enzyme");
import {GridItem} from '../../components/GridItem';

describe('Pruebas a Grid Item', ()=>{

    const title = 'Un Titulo';
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GridItem
        title={title}
        url={url}
    />)

    test('Renderizado del componente', ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Validar valores correctos', ()=>{

        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(title);

    });

    test('Validar imagen', ()=>{

        const img = wrapper.find('img');
        expect(img.props().src).toBe(url)


    });

    test('Debe tener una clase equis', ()=>{
        const div = wrapper.find('div');
        
        const className = div.prop('className');
        
        expect(className.includes('card')).toBe(true)
    })

});