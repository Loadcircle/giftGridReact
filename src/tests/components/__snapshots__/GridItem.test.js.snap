// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a Grid Item Renderizado del componente 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
